src/java.base/share/classes/java/lang/Class.java

```diff
@@ -651,7 +654,9 @@ public boolean isIdentity() {
      * @since Valhalla
      */
     public boolean isValue() {
-        return (this.getModifiers() & Modifier.VALUE) != 0;
+        return ValhallaFeatures.isEnabled() &&  // Before Valhalla no classes are value classes
```

Comment:
`isValue` does not need to check `-XX:+EnableVahalla` since the new `VALUE` flag can only be set in a value class and if the feature is enabled.